scripts/zephyr_module.py

```diff
@@ -177,6 +207,9 @@ def main():
     parser.add_argument('--cmake-out',
                         help="""File to write with resulting <name>:<path>
                              values to use for including in CMake""")
+    parser.add_argument('--settings-out',
```

Comment:
This looks identical in purpose to --cmake-out; do we really need a separate file? If so, why?

Reply:
Yes, we should keep those separated.

Although both files are currently used only by CMake, then the cmake out file contains module name and path for inclusion into the build tree itself, with `add_subdirectory` (They contain the path to the CMakeLists.txt file).
And therefore is very CMake specific.

The settings file is simply a file containing build settings from each Zephyr module.
This could be useful for other things in future, for example documentation generation, if we want to create a page with boards located at other Zephyr modules.

But of course, CMake will use the settings for building 😉 

And then finally, settings file is a `key:value` pair in all lines, but in `--cmake-out` file then all lines in  are values (The keys are `module_name:path`).